ansible_runner/utils/streaming.py

```diff
@@ -19,14 +20,21 @@ def stream_dir(source_directory, stream):
                     if relpath == ".":
                         relpath = ""
                     for fname in files:
-                        fullpath = os.path.join(dirpath, fname)
-                        if not os.path.isfile(fullpath):
-                            raise FileNotFoundError(
-                                "invalid symlink found; file={0}.".format(fullpath)
+                        full_path = os.path.join(dirpath, fname)
+                        # Magic to preserve symlinks
+                        if os.path.islink(full_path):
+                            archive_relative_path = os.path.relpath(dirpath, source_directory)
+                            file_relative_path = os.path.join(archive_relative_path, fname)
+                            zip_info = zipfile.ZipInfo(file_relative_path)
+                            zip_info.create_system = 3
+                            permissions = 0o777
```

Comment:
Do unzipped files result in permission `777` ?

Reply:
See my comment below.